addons/website_sale/models/website_snippet_filter.py

```diff
@@ -196,3 +196,11 @@ def _get_products_alternative_products(self, website, limit, domain, context):
                     display_default_code=False,
                 ).search(domain, limit=limit)
         return products
+
+    def _prepare_values(self, limit=None, search_domain=None):
+        website = self.env['website'].get_current_website()
+        if (website.ecommerce_access == 'logged' and self.env.user._is_public()
```

Comment:
```suggestion
        if (not website.has_shop_access() and
```

Reply:
Same case, has_shop_access cannot be used here.